ingestion/functions/parsing/japan/japan.py

```diff
@@ -43,23 +41,44 @@ def convert_location(raw_entry):
         if term and term != "Unspecified"]
     return {"query":  ", ".join(query_terms)}
 
-
 def detect_notes(raw_notes):
     if "notes" in raw_notes:
         return raw_notes["notes"]
     else:
         return None
 
-
-def convert_date(raw_date: str):
+def convert_date(raw_date: Dict):
     """
     Convert raw date field into a value interpretable by the dataserver.
-
+​
     The date filtering API expects mm/dd/YYYYZ format.
     """
-    date = datetime.strptime(raw_date, "%Y-%m-%d")
+    date = datetime.strptime(raw_date["dateAnnounced"], "%Y-%m-%d")
     return date.strftime("%m/%d/%YZ")
 
+def convert_additional_sources(additional_source_url: Dict):
+    sources = []
+    if "sourceURL" in additional_source_url:
+      sources.append({"sourceUrl": additional_source_url["sourceURL"]})
+    if "deathSourceURL" in additional_source_url:
+      sources.append({"sourceUrl": additional_source_url["deathSourceURL"]})
+    return sources or None
+
+def convert_outcome(raw_outcome: Dict, raw_death_date: Dict):
+    if "patientStatus" in raw_outcome:
+        if raw_outcome["patientStatus"] == "Deceased":
+            death_date = datetime.strptime(raw_death_date["deceasedDate"], "%Y-%m-%d")
+            return {"name": "outcome",
+                    "dateRange":{
+                        "start": death_date.strftime("%m/%d/%YZ"),
+                        "end": death_date.strftime("%m/%d/%YZ")
```

Comment:
As before, you could probably fix it so that convert_date() handles this as well. Up to you!